scapy/layers/rtp.py

```diff
@@ -51,7 +51,7 @@ class RTP(Packet):
     ]
 
     def guess_payload_class(self, pkt):
-        if ("extension" in self.fields and self.fields["extension"]==1):
+        if ("extension" in self.fields and self.extension):
```

Comment:
I'm not sure to understand what you are trying to do here. Wouldn't `if self.extension:` be enough?

In that case, you could rewrite the method as:
```Python
    def guess_payload(self, pkt):
        return RTPExtension if self.extension else conf.raw_layer
```

Actually, even better, wouldn't you get the exact same result if you removed this method and just added a `bind_layer()` call?